src/radical/pilot/agent/scheduler/base.py

```diff
@@ -286,7 +293,7 @@ def create(cls, cfg, session):
         if cls != AgentSchedulingComponent:
             raise TypeError("Scheduler Factory only available to base class!")
 
-        name = cfg['scheduler']
+        name = session._reg['rcfg.agent_scheduler']
```

Comment:
same as for `spawner` (but looks like need to set a default value here)

https://github.com/radical-cybertools/radical.pilot/blob/a4caa5e6b23b94b99baa329b02bf33dacdf9dab8/src/radical/pilot/pmgr/launching/base.py#L567

https://github.com/radical-cybertools/radical.pilot/blob/a4caa5e6b23b94b99baa329b02bf33dacdf9dab8/src/radical/pilot/pmgr/launching/base.py#L860

Reply:
Same.